sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/rows.scala

```diff
@@ -201,7 +201,7 @@ class GenericRow(protected[sql] val values: Array[Any]) extends Row {
 
   override def toSeq: Seq[Any] = values.toSeq
 
-  override def copy(): Row = this
+  override def copy(): Row = new GenericRow(values.clone())
```

Comment:
You cannot a put value back in the ```GenericRow``` when you have taken it out, unless it is a mutable object. Could you provide an example for this? Which is also a good basis for a unit test.

```copy``` is merely a contract all rows need to adhere to. If a row is inmutable, why copy it? It also avoids a lot object allocations. I think ```GenericInternalRow``` should also return reference to it self on ```copy```.